### ISSUE TITLE
Miniconda: `conda activate` fails on freshly-created environment: (eval):10: unmatched '

### ISSUE THREAD

**Issue by jarppiko** [conda/conda#12763]
### What happened?

I am trying on install Miniconda and create an environment. Everything goes fine until I try to activate the environment using `conda activate env_name`. This results an error: 

### Zsh
```
(eval):10: unmatched '
```

### Bash
```
bash: syntax error near unexpected token `('
```

## Details

* I have installed Miniconda  using [latest installation instructions](https://docs.conda.io/projects/conda/en/latest/user-guide/install/linux.html). I have tried both Python 3.8 and 3.10 versions. 
* I have chose to `init` conda in the end of the installation process. I have tried manual `init` too, but it did not help
* The installation does not yield any errors
* Creating an environment  goes fine
* Activation fails with the error messages above.

### Conda Details

<details>
<summary><code>conda info</code></summary>

```shell
active environment : None
            shell level : 0
       user config file : /home/user/.condarc
 populated config files : /home/user/.condarc
          conda version : 23.5.0
    conda-build version : not installed
         python version : 3.8.16.final.0
       virtual packages : __archspec=1=x86_64
                          __cuda=12.2=0
                          __glibc=2.35=0
                          __linux=5.15.90.1=0
                          __unix=0=0
       base environment : /home/user/miniconda3  (writable)
      conda av data dir : /home/user/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/user/miniconda3/pkgs
                          /home/user/.conda/pkgs
       envs directories : /home/user/miniconda3/envs
                          /home/user/.conda/envs
               platform : linux-64
             user-agent : conda/23.5.0 requests/2.29.0 CPython/3.8.16 Linux/5.15.90.1-microsoft-standard-WSL2 ubuntu/22.04.2 glibc/2.35
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

</details>

<details>
<summary><code>conda config</code></summary>

```shell
==> /home/user/.condarc <==
auto_activate_base: False
```

</details>

<details>
<summary><code>conda list</code></summary>

### Conda list

```shell
# packages in environment at /home/user/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             5.1                       1_gnu    defaults
boltons                   23.0.0           py38h06a4308_0    defaults
brotlipy                  0.7.0           py38h27cfd23_1003    defaults
ca-certificates           2023.01.10           h06a4308_0    defaults
certifi                   2023.5.7         py38h06a4308_0    defaults
cffi                      1.15.1           py38h5eee18b_3    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
conda                     23.5.0           py38h06a4308_0    defaults
conda-content-trust       0.1.3            py38h06a4308_0    defaults
conda-package-handling    2.1.0            py38h06a4308_0    defaults
conda-package-streaming   0.8.0            py38h06a4308_0    defaults
cryptography              39.0.1           py38h9ce1e76_0    defaults
idna                      3.4              py38h06a4308_0    defaults
jsonpatch                 1.32               pyhd3eb1b0_0    defaults
jsonpointer               2.1                pyhd3eb1b0_0    defaults
ld_impl_linux-64          2.38                 h1181459_1    defaults
libffi                    3.4.4                h6a678d5_0    defaults
libgcc-ng                 11.2.0               h1234567_1    defaults
libgomp                   11.2.0               h1234567_1    defaults
libstdcxx-ng              11.2.0               h1234567_1    defaults
ncurses                   6.4                  h6a678d5_0    defaults
openssl                   1.1.1t               h7f8727e_0    defaults
packaging                 23.0             py38h06a4308_0    defaults
pip                       23.0.1           py38h06a4308_0    defaults
pluggy                    1.0.0            py38h06a4308_1    defaults
pycosat                   0.6.4            py38h5eee18b_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyopenssl                 23.0.0           py38h06a4308_0    defaults
pysocks                   1.7.1            py38h06a4308_0    defaults
python                    3.8.16               h7a1cb2a_3    defaults
readline                  8.2                  h5eee18b_0    defaults
requests                  2.29.0           py38h06a4308_0    defaults
ruamel.yaml               0.17.21          py38h5eee18b_0    defaults
ruamel.yaml.clib          0.2.6            py38h5eee18b_1    defaults
setuptools                65.6.3           py38h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
sqlite                    3.41.2               h5eee18b_0    defaults
tk                        8.6.12               h1ccaba5_0    defaults
toolz                     0.12.0           py38h06a4308_0    defaults
tqdm                      4.65.0           py38hb070fc8_0    defaults
urllib3                   1.26.15          py38h06a4308_0    defaults
wheel                     0.38.4           py38h06a4308_0    defaults
xz                        5.4.2                h5eee18b_0    defaults
zlib                      1.2.13               h5eee18b_0    defaults
zstandard                 0.19.0           py38h5eee18b_0    defaults
```

</details>

### Additional Context

* The OS is Ubuntu 22.04.2 running on Windows 11 (WLS2)
* Python itself works fine. I use normally Python virtual environments for development, but decided to give a try to `conda` as an "easy" way to manage packages.
* I use Zsh as the login shell
* Zsh has [Oh-My-Zsh](https://ohmyz.sh/) installed

**Comment by kenodegard** [comment]
@jarppiko can you provide the output when you run `conda shell.posix activate env_name`?

**Comment by jarppiko** [comment]
Thank you for reacting so quickly. Here is to output.  

```
% conda shell.posix activate nerfstudio
PS1='(nerfstudio) '
export PATH='/home/user/miniconda3/envs/nerfstudio/bin:/home/user/miniconda3/condabin:/home/user/.local/bin:/home/user/go/bin:/usr/local/cuda-11.8/bin:/home/user/.local/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/mnt/c/Program Files/WindowsApps/Microsoft.WindowsTerminal_1.16.10262.0_x64__8wekyb3d8bbwe:/mnt/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.8/bin:/mnt/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.8/libnvvp:/mnt/c/VulkanSDK/1.3.250.0/Bin:/mnt/c/Program Files (x86)/Razer Chroma SDK/bin:/mnt/c/Program Files/Razer Chroma SDK/bin:/mnt/c/Windows/system32:/mnt/c/Windows:/mnt/c/Windows/System32/Wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0:/mnt/c/Windows/System32/OpenSSH:/mnt/c/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64:/mnt/c/Program Files/NVIDIA Corporation/NVIDIA NvDLISR:/mnt/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/mnt/c/Users/user/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/user/AppData/Local/GitHubDesktop/bin:/mnt/c/Users/user/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/bin:/mnt/c/Program Files/Multipass/bin:/mnt/c/Program Files/dotnet:/mnt/c/Program Files/Go/bin:/mnt/c/Program Files (x86)/Microsoft SQL Server/150/DTS/Binn:/mnt/c/Program Files/Azure Data Studio/bin:/mnt/c/WINDOWS/system32:/mnt/c/WINDOWS:/mnt/c/WINDOWS/System32/Wbem:/mnt/c/WINDOWS/System32/WindowsPowerShell/v1.0:/mnt/c/WINDOWS/System32/OpenSSH:/mnt/c/Program Files/PuTTY:/mnt/c/Program Files (x86)/Bitvise SSH Client:/mnt/c/Program Files/Microsoft SQL Server/150/Tools/Binn:/mnt/c/Program Files/Git/cmd:/mnt/d/nerf/COLMAP-3.8-windows-cuda/bin:/mnt/c/Program Files/NVIDIA Corporation/Nsight Compute 2022.3.0:/mnt/c/Users/user/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/user/AppData/Local/GitHubDesktop/bin:/mnt/c/Users/user/AppData/Local/Programs/Microsoft VS Code/bin:/mnt/c/Users/user/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0/LocalCache/local-packages/Python39/Scripts':/mnt/c/rtools40:/mnt/c/Users/user/go/bin:/mnt/c/Program Files/Azure Data Studio/bin:/mnt/c/Users/user/.dotnet/tools:/home/user/.local/bin'
export CONDA_PREFIX='/home/user/miniconda3/envs/nerfstudio'
export CONDA_SHLVL='1'
export CONDA_DEFAULT_ENV='nerfstudio'
export CONDA_PROMPT_MODIFIER='(nerfstudio) '
export CONDA_EXE='/home/user/miniconda3/bin/conda'
export _CE_M=''
export _CE_CONDA=''
export CONDA_PYTHON_EXE='/home/user/miniconda3/bin/python'
```

**Comment by kenodegard** [comment]
@jarppiko your `PATH` has an extra `'` near the end, please review how you define your `PATH` prior to using conda, removing the extra `'` will resolve the issue:

```
export PATH='...LocalCache/local-packages/Python39/Scripts':/mnt/c/rtools40...'
                                                          ^
```

**Comment by jarppiko** [comment]
🤦‍♂️  

Embarrassing.... I'm really sorry for bothering you with this. I found it so odd that both `zsh` and `bash` had issues, but everything else was working all fine. I spent hours trying different versions and trying to add debug messages to the `__conda_activate` shell function call without success. 

This problem was actually on Windows System environment variable side that are for some reason exported to WSL side. It has been there years. 

Big thanks 🙏
